### PR TITLE
Corrects typos in reserved characters

### PR DESCRIPTION
**Pull request by jfinkels**
Also clarifies difference between comma and period by using parallel syntax.

### CODE REVIEW

**Comment by ethanresnick** [comment]
Thanks @jfinkels! 

I agree we should clean this language up. If we're on this path though, we might as well go all the way, so I've made an inline comment for how to make the language even shorter/clearer. Let me know what you think of that!

File: format/index.md [hunk 1]
```diff
@@ -589,9 +589,9 @@ first or last character:
 The following characters **MUST NOT** be used in member names:
 
 - U+002B PLUS SIGN, "+" _(used for ordering)_
-- U+002C COMMA, "," _(used separator for multiple relationship paths)_
-- U+002E PERIOD, "." _(used as relationship path separators)_
-- U+005B LEFT SQUARE BRACKET, "[" _(use in sparse fieldsets)_
+- U+002C COMMA, "," _(used as a separator between relationship paths)_
```
Comment (ethanresnick) [inline format/index.md:592]: How about just "used to separate relationship paths"?

File: format/index.md [hunk 2]
```diff
@@ -589,9 +589,9 @@ first or last character:
...
+- U+002E PERIOD, "." _(used as a separator within relationship paths)_
```
Comment (ethanresnick) [inline format/index.md]: How about "used as a separator within **a** relationship path"

File: format/index.md (hunk 1 above)
Comment (jfinkels) [inline format/index.md:592]: To me, I think that's still ambiguous about separating _within_ versus separating _between_. If you do "used to separate relationship paths" for comma, I guess you should do "used to separate relationship names within relationship paths" for period.

Comment (ethanresnick) [inline format/index.md:592]: > To me, I think that's still ambiguous about separating within versus separating between.

Ok, fair. I've merged your language as is :)

**Comment by jfinkels** [comment]
Cool!

**Pull request merged**